### DISCUSSION TITLE
how can i run this on the browser offline?

### DISCUSSION THREAD

**Discussion by dannywynn** [pymupdf/PyMuPDF#4191] [Installation Topics]
i have a script that i created using this package 
i want to use that script in the browser (offline) 

the pdfs are confidential and user wants to do all the parsing offline on his pc
is there any way? 

thank you.

**Comment by JorjMcKie** [comment]
This doesn't make sense:
Somewhere, Python must exists and contain the PyMuPDF installation. Either you use Python and PyMuPDF installed on a server like [here](https://pymupdf.io), or you you must have installed Python locally - with PyMuPDF inside.
A browser not accessing internet resources must use local resources on your machine.

**Comment by dannywynn** [comment]
> This doesn't make sense: Somewhere, Python must exists and contain the PyMuPDF installation. Either you use Python and PyMuPDF installed on a server like [here](https://pymupdf.io), or you you must have installed Python locally - with PyMuPDF inside. A browser not accessing internet resources must use local resources on your machine.

can PyMuPDF be compiled to WebAssembly, if not, are there any alternatives for working with PDFs offline in a browser?

**Comment by JorjMcKie** [comment]
Why does your user not simply install Python + PyMuPDF on his machine? What's so cool to use a browser as the running platform?

The above link to [PyMuPDF online](https://pymupdf.io/) is a WebAssembly installation of PyMuPDF. It is based on Pyodide. So in principle this can be done, but we haven't published this as an official option. You (/ your user) definitely must install [Pyodide](https://pyodide.org/en/stable/) in a way that runs locally.
Maybe @julian-smith-artifex-com has more comments on this.

I actually don't understand the basic motivation: is it to avoid installing Python? Why don't you compile your app to an executable?
